[docathon] Configuring assets guide (#23924)
dagster-io/dagster: docs/docs-beta/docs/guides/data-modeling/configuring-assets.md

---

> title: Configuring assets and ops via the UI

Reviewer: ```suggestion
title: Configuring runs via the UI
```

Reviewer 2: ```suggestion
title: Configuring assets and ops using the Dagster UI
```

Nit - we're moving away from `via` (see the Vale rule) and when referring to the UI in a page title, it should be the complete `Dagster UI`

sidebar_label: Configuring assets via the UI
sidebar_position: 50
---

You will commonly want to manually materialize assets using the Dagster UI to backfill historical data, debug a production issue, or some other one-off task.

Often, you will want to be able to tweak some parameters when materializing these assets. This can be accomplished through the asset configuration system.

## What you'll learn

> - How to add configuration to your assets

Reviewer: This makes me think the section is about configuring the assets, when, if I understand correctly, it's more about "How to define configuration parameters for your assets that can be set when launching a run that targets them".

That's a bit of a mouthful though. Maybe "How to add run configuration schema to your assets"?

Reviewer 3: I agree with Sandy, but I think 'run configuration schema' is more about the 'machinery' than the goal.
Maybe something like 'Configuring Assets when Launched'?

Reviewer 2: "Configuring assets when launching runs" or "a run"?

- How to modify the configuration when launching a run

> - When to use asset configuration vs. [Resources](/docs/concepts/resources)

Reviewer 2: ```suggestion
- When to use asset configuration vs. [resources](/docs/concepts/resources)
```

Nit - this doesn't need to be capitalized, as it's not a proper noun

---

<details>
  <summary>Prerequisites</summary>

To follow the steps in this guide, you'll need:

> - A basic understanding of Dagster and assets. See the [Quick Start](/tutorial/quick-start) tutorial for an overview.

Reviewer 2: Because this guide also talks about ops, should those be in the prereqs somewhere?

> - Passing familiarity with [Pydantic](https://docs.pydantic.dev/latest/)

Reviewer 2: What is 'passing familiarity' as opposed to just 'familiarity'?

</details>

---

> ## Adding configuration to your asset

Reviewer 2: ```suggestion
## Adding configuration to assets
```

Let's remove `your` from section headings - it cuts down on words/length and could potentially help with search results. Maybe. 🤷‍♀️

> You must define a schema for the configuration you want to attach to your asset. For example, let's say we want to allow users to change a parallelism parameter for an asset:

Reviewer 3: ```suggestion
For an asset to be configurable, you must first define a schema that inherits from the Dagster `Config` class. For example, let's say we want to allow users to change a parallelism parameter for an asset:
```

<CodeExample filePath="guides/data-modeling/configuring-assets/config-schema.py" language="python" title="Adding configuration" />

## Modifying the configuration when launching a run

> When launching a run using Dagster's Launchpad, you can provide a run config file as YAML or JSON that overrides the default configuration for your asset:

Reviewer 3: Very annoying that this example leads with `ops` when we are talking about assets. Wondering if we mention something like 

Run configurations reference an `op` which is the underlying computation of an asset. For more information, see `[Assets vs Ops](/todo)`
